### DISCUSSION TITLE
Unable to link to libsodium in VS2019

### DISCUSSION THREAD

**Discussion by NE1W01F** [jedisct1/libsodium#1183] [Q&A]
I hate to ask but I have been having problems linking libsodium.

The libsodium version is libsodium-1.0.18-stable-msvc I got it from the website, I was also using the debug static linked lib. I also added the `#define SODIUM_STATIC` to my main cpp file. But I still get a linking error. I have been trying to fix this for an hour now. I have googled the problem and searched this github to.

The code I used
```cpp
#include <sodium.h>
#define SODIUM_STATIC

int main(){
     return sodium_init();
}
```

The errors I got
```
LNK2019	unresolved external symbol __imp__sodium_init referenced in function _main
LNK1120	1 unresolved externals
```

The only settings i changed in my project was
`Configuration Properties -> Linker -> Additional Dependencies = libsodium.lib`
`Configuration Properties -> Linker -> Additional Library Directories`
`Configuration Properties -> C/C++ -> Additional Include Directories`
`Configuration Properties -> C/C++ -> Runtime Library = Multi-Threaded Debug`

If you have any ideas on how to fix this problem then please repliy.

**Comment by ipduffy** [comment]
Put your #define SODIUM_STATIC before the #include <sodium.h>
